src/client/Microsoft.Identity.Client/ILongRunningWebApi.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Methods for long running processes in web APIs
+    /// </summary>
+    public interface ILongRunningWebApi
+    {
+        /// <summary>
+        /// Acquires an access token for this web API from the authority configured in the application,
+        /// in order to access another downstream protected web API on behalf of a user using the OAuth 2.0 On-Behalf-Of flow.
+        /// See https://aka.ms/msal-net-on-behalf-of.
+        /// This confidential client application was itself called with a token which will be provided in the
+        /// <paramref name="userToken">userToken</paramref> parameter.
+        /// </summary>
+        /// <param name="scopes">Scopes requested to access a protected API</param>
+        /// <param name="userToken">A JSON Web Token which was used to call the web API and contains the credential information
+        /// about the user on behalf of whom to get a token.</param>
+        /// <param name="longRunningProcessSessionKey">Key by which to look up the token in the cache.
```

Comment:
@pmaytak - let's add `sid` claim as the suggested key to use, and add a remark explaining that MSAL cannot read tokens on its own.

Reply:
@bgavrilMS is it correct to read this as "MSAL performs the OBO without validating who the token is for"? I'd suggest we vet this path for confused deputy attack surface.

Reply:
@hpsin - MSAL doesn't look at incoming token at all. It's up to the web api to perform any validations. I believe Ms.Id.Web does some of that.

I may not understand the question, but allow me to elaborate. 

Today, we advise people wanting to implement the "long running process OBO" scenario to keep around the `incoming_assertion`. This is to avoid the "confused deputy" attack, i.e. the AT / RT from OBO can only be retrieved if the original assertion is found. However, this is pretty complicated, and we end up asking folks to keep around an expired JWT. They mostly keep it around in their own cache as "username" -> "original_jwt".  

`AcquireTokenInLongRunningProcess(scopes, key)` aims to fix that. It is a slightly more flexible version of `AcquireTokenOnBehalfOf(scopes, incomming_assertion)`, to allow the web api to let go of the expired incomming_assertion and to remember only the `sid`. Internally, MSAL will locate the AT / RT in its caches by this key. 

Shall we meet to discuss this in more detail?

Reply:
Thanks, I think that answers my question in general. Do we have any kind of ratchet in place that tries to ensure that the web api has performed validation?

Reply:
> @pmaytak - let's add `sid` claim as the suggested key to use, and add a remark explaining that MSAL cannot read tokens on its own.

We cannot crack-open the token, so the claim would have to be provided by the application to MSAL.NET. Also, `sid` is an optional claim, This requires an aka.ms link.